File: src/types/order.ts
export interface Order {
  orderId: number;
  userId: number;
  totalAmount: number;
  status: OrderStatus;
  payType?: PaymentType;
  payTime?: string;
  createdAt: string;
  updatedAt: string;
  items: OrderItem[];
}

export type OrderStatus = 'pending' | 'paid' | 'cancelled' | 'completed';
export type PaymentType = 1 | 2 | 3; // 1: alipay, 2: wechat, 3: credit card

export interface OrderItem {
  orderItemId: number;
  productId: number;
  quantity: number;
  price: number;
  productName: string;
  productImage?: string;
}

export interface CreateOrderRequest {
  products: Array<{
    productId: number;
    quantity: number;
  }>;
}

export interface OrderResponse {
  code: string;
  info: string;
  data: Order;
}

export interface OrderListResponse {
  code: string;
  info: string;
  data: Order[];
}

File: backend/app/schemas/order.py
from pydantic import BaseModel


class OrderItemOut(BaseModel):
    orderItemId: int
    productId: int
    skuId: int | None = None
    skuName: str | None = None
    productName: str
    price: float
    quantity: int
    imageUrl: str | None = None


class OrderOut(BaseModel):
    orderId: int
    userId: int
    totalAmount: float
    payableAmount: float
    deliveryFee: float
    deliveryType: str
    paymentType: str | None = None
    status: str
    address: dict | None = None
    remark: str
    paidAt: str | None = None
    createdAt: str
    updatedAt: str
    items: list[OrderItemOut] = []


class CreateOrderRequest(BaseModel):
    addressId: int
    deliveryType: str = "standard"
    paymentType: str = "alipay"
    remark: str = ""


class PayOrderRequest(BaseModel):
    paymentType: str = "alipay"
